File: dcp-solutions/2025-02-03/solution.ex
```elixir
defmodule BinaryTreePruner {
  def prune(root) do
    # Base case: empty tree or node with value 1
    if root == nil || root.val == 1, do: root
    
    # Recursively check left and right subtrees
    new_left = prune(root.left)
    new_right = prune(root.right)
    
    # If both subtrees contain only 0s, return nil
    if (new_left == nil && new_right == nil), do: nil
    
    # Otherwise, update the current node's value and connect it to the pruned subtrees
    %{root => %{
      val: root.val,
      left: new_left,
      right: new_right
    }}
  end

defmodule BinaryTreePruner {
  def traverse(root) do
    case root do
      nil -> []
      %{val: val, left: left, right: right} ->
        [val | traverse(left)] ++ traverse(right)
    end
  end
}

# Example usage:
root = %{
  val: 0,
  left: %{
    val: 1,
    left: %{
      val: 0,
      left: %{},
      right: %{}
    },
    right: %{
      val: 1,
      left: %{},
      right: %{}
    }
  },
  right: %{
    val: 0,
    left: %{
      val: 0,
      left: %{},
      right: %{}
    },
    right: %{
      val: 0,
      left: %{},
      right: %{}
    }
  }
}

pruned = BinaryTreePruner.prune(root)

IO.inspect(BinaryTreePruner.traverse(pruned))
```

File: dcp-solutions/2025-02-03/solution.py
```python
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None

def pruneTree(root):
    return prunetree_recursive(root)

def prunetree_recursive(node):
    if not node:
        return False
    
    is_zero = node.val == 0 and (not node.left or not prunetree_recursive(node.left)) and not node.right or not prunetree_recursive(node.right)
    
    if is_zero:
        return None
    else:
        node.left = prunetree_recursive(node.left)
        node.right = prunetree_recursive(node.right)
        
        return node
```